### ISSUE TITLE
Ledger Reconcile Can't Find Included Files

### ISSUE THREAD

**Issue by sdubinsky** [ledger/ledger-mode#427]
Emacs version: 29.3
ledger version: 3.2.1-20200518
ledger-mode version: 20240537.2134

When I try to reconcile a file that includes another file, it errors out with: `While parsing file "", line 5: Error: File to include was not found: "reconcile2.dat"`

I ran this under emacs -Q to be sure.

reconcile1.dat:
```
include ./reconcile2.dat

2024-01-01 test
    Test1                                         $1
    Test3
```

reconcile2.dat:
```
2024-01-01 test2
    Test3                                         $3
    Test4
```

I tried it both with and without the ./ in the pathname.

**Comment by sdubinsky** [comment]
Oh oops, that works too.

**Comment by enderw88** [comment]
This is not a bug. Just a feature I never implemented. It requires a lot
more work than it’s worth to me. No one has ever wanted to tackle it.

Craig, Corona De Tucson, AZ
[image: missile_flyout]
enderw88.wordpress.com

**Comment by sdubinsky** [comment]
Oh ok, fair enough.

For anyone else, my workaround is just to comment that line out before i do a reconcile.

**Comment by bcc32** [comment]
@sdubinsky does it look like you might have the issue in #329?  Maybe you just need to upgrade ledger itself?

**Comment by sdubinsky** [comment]
No, it works in ledger itself.

**Comment by bcc32** [comment]
Sorry, I'm not sure I understand your last comment.  Just to clarify, the issue in #329 is that (the bad version of) ledger doesn't handle `include` properly when called like so:

```
ledger b -f - <reconcile1.dat
```

whereas the buggy build of ledger still behaves properly when running:

```
ledger b -f reconcile1.dat
```

Does your version of ledger behave properly when running the above commands?

**Comment by sdubinsky** [comment]
Yes.  As I said, my files work in ledger itself, they only break in ledger-mode.

**Comment by bgainor** [comment]
I'm the reporter of #329, and FWIW this was fixed for me when I upgraded ledger to 3.3.2. I had a similar problem, where the file was being correctly included when I ran `ledger` from the command line, but not from `ledger-mode` in Emacs. It's worth noting that my include statement _doesn't_ have the `./`; I'm just including the basename of the file (which is in the same directory as my main ledger file), so if upgrading still doesn't fix it for you, you might try that.
